public/js/till.js

```diff
@@ -0,0 +1,76 @@
+function Till(details){
+  this.details = details;
+  this.taxes = 0;
+  this.total = 0;
+  this.specialDiscount = 0.1;
+  this.discountRate = 0.05;
+  this.taxRate = 0.0864;
+  this.order = {};
+  this.receipt = {};
+};
+
+
+Till.prototype.calculateTotal = function(item) {
```

Comment:
Could the total be calculated in a seperate object, maybe a calculator object that is responsible for doing all the numerical things? As alway you want to be thinking about the Single Responsibility Principle and breaking out your objects into other objects when you find your objects getting too complex